src/main/java/org/jabref/logic/integrity/ValidBibtexKeyChecker.java

```diff
@@ -18,7 +18,16 @@ public ValidBibtexKeyChecker(boolean enforceLegalKey) {
 
     @Override
     public Optional<String> checkValue(String value) {
+
+        // Fix #2: BibtexKeyGenerator.cleanKey() does not accept a null value
+        // for the "value" parameter. The 'if' statement below is added to check for
+        // both a null and zero length string and to add the empty key warning.
+        if ((value == null) || (value.length() == 0)) {
```

Comment:
`StringUtil.isBlank(...)` is slightly prefered because then also whitespace is correctly recognized as a missing key.